aikau/src/main/resources/alfresco/services/CloudSyncService.js

```diff
@@ -0,0 +1,568 @@
+       */
+      i18nRequirements: [{i18nFile: "./i18n/CloudSyncService.properties"}],
+      
+      /**
+       * Sets up the subscriptions for the CommentService
+       * 
+       * @instance
+       * @listens module:alfresco/core/topics#CLOUD_AUTHENTICATION_REQUEST
+       * @listens module:alfresco/core/topics#GET_CLOUD_TENANTS
+       * @listens module:alfresco/core/topics#GET_CLOUD_SITES
+       * @listens module:alfresco/core/topics#INIT_CLOUD_SYNC
+       * @listens module:alfresco/core/topics#SYNC_TO_CLOUD
+       */
+      registerSubscriptions: function alfresco_services_CloudSyncService__registerSubscriptions() {
+         this.alfSubscribe(topics.CLOUD_AUTHENTICATION_REQUEST, lang.hitch(this, this.onCloudAuthentication));
+         this.alfSubscribe(topics.GET_CLOUD_PATH, lang.hitch(this, this.getTenantSitePath));
+         this.alfSubscribe(topics.GET_CLOUD_SITES, lang.hitch(this, this.getTenantSiteOptions));
+         this.alfSubscribe(topics.GET_CLOUD_TENANTS, lang.hitch(this, this.getTenantOptions));
+         this.alfSubscribe(topics.INIT_CLOUD_SYNC, lang.hitch(this, this.onCloudSyncRequest));
+         this.alfSubscribe(topics.SYNC_TO_CLOUD, lang.hitch(this, this.onCloudSyncConfirmation));
+      },
+
+      /**
+       * This is provided for the purpose of generating a list of the tenants that the user has 
+       * access to.
+       *
+       * @instance
+       * @param  {object} payload The options request payload
+       */
+      getTenantOptions: function alfresco_services_CloudSyncService__getTenantOptions(payload) {
+         this.serviceXhr({
+            url : AlfConstants.PROXY_URI + "cloud/tenant/information",
+            method: "GET",
+            data: payload,
+            successCallback: this.processTenantOptions,
+            callbackScope: this
+         });
+      },
+
+      /**
+       * This is the success call back for [getTenantOptions]{@link module:alfresco/services/CloudSyncService#getTenantOptions}.
+       * It processes the raw data to create an array of the tenants that the user has permission to 
+       * sync content to.
+       * 
+       * @instance
+       * @param {object} response The response from the request
+       * @param {object} originalRequestConfig The configuration passed on the original request
+       */
+      processTenantOptions: function alfresco_services_CloudSyncService__processTenantOptions(response, originalRequestConfig) {
+         var options = [];
+
+         // Put the home tenant as the first option...
+         var homeTenant = lang.getObject("homeTenant.name", false, response);
+         options.push({
+            label: homeTenant,
+            value: homeTenant
+         });
+
+         // Add any secondary tenants that have sync-enabled...
+         var secondaryTenants = lang.getObject("secondaryTenants", false, response);
+         array.forEach(secondaryTenants, function(secondaryTenant) {
```

Comment:
`dojo/_base/array` actually handles falsy values, but it might be better to maybe have `secondaryTenants || []` just for clarity?